### PR TITLE
Add Tasks and Comments entities

### PR DESCRIPTION
**Pull request by kirill-oleynik**

### CODE REVIEW

File: app/concepts/api/v1/comments/operation/create.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      context['contract.default'].success?
+    end
+
+    def find_user(context, token:, **)
+      user_id = JwtSession::GetUserIdFromToken.new(token).call
+      context[:user] = User.find_by(id: user_id)
+    rescue JWTSessions::Errors::Unauthorized
+      context[:operation_status] = :unauthorized
+      false
+    end
```
Comment (StanisLoveSid) [inline app/concepts/api/v1/comments/operation/create.rb:21]: could be moved to Subprocess operation
`payload` from JWT sessions can be used here

File: app/concepts/api/v1/comments/operation/update.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  # Update Task Comment entity operation
+  class Update < ApplicationOperation
+    step :call_contract
+    step :find_user
+    step :update_comment
+
+    def call_contract(context, params:, **)
+      context['contract.default'] = API::V1::Comments::Contract::SetAttributes.new.call(params[:comment])
+      context['contract.default'].success?
+    end
```
Comment (StanisLoveSid) [inline app/concepts/api/v1/comments/operation/update.rb:13]: should be used default validation macroses here

File: config/initializers/shrine.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    secret_access_key: Rails.application.credentials.aws[:secret_access_key],
+  }
+  Shrine.storages = {
+    cache: Shrine::Storage::S3.new(prefix: 'cache', **s3_options),
+    store: Shrine::Storage::S3.new(**s3_options),
+  }
+when 'development'
+  Shrine.storages = {
+    cache: Shrine::Storage::FileSystem.new('storage', prefix: 'uploads/cache'),
+    store: Shrine::Storage::FileSystem.new('storage', prefix: 'uploads'),
```
Comment (StanisLoveSid) [inline config/initializers/shrine.rb:21]: move to env files

File: config/initializers/shrine.rb [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    cache: Shrine::Storage::Memory.new,
+    store: Shrine::Storage::Memory.new
+  }
+end
+
+Shrine.plugin :activerecord
+Shrine.plugin :cached_attachment_data
+Shrine.plugin :restore_cached_data
+Shrine.plugin :validation_helpers
+Shrine.plugin :determine_mime_type
```
Comment (StanisLoveSid) [inline config/initializers/shrine.rb:34]: move to uploader please

File: spec/factories/tasks.rb [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+    f.deadline { (Date.today + rand(100)).to_s }
+    user
+  end
+end
+
+def create_task_with_comments(comments_count: 5)
+  FactoryBot.create(:task) do |task|
+    FactoryBot.create_list(:comment, comments_count, task_id: task.id)
+  end
+end
```
Comment (StanisLoveSid) [inline spec/factories/tasks.rb:16]: move to trait please

File: spec/models/task_spec.rb [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+  end
+
+  it 'destroy dependent comments after deletion' do
+    expect(Comment.count).to equal(0)
+    task = create_task_with_comments(comments_count: 10)
+    expect(Comment.count).to equal(10)
+    task.destroy
+    expect(Comment.count).to equal(0)
+  end
+end
```
Comment (StanisLoveSid) [inline spec/models/task_spec.rb:16]:
```ruby
RSpec.describe example, type: :model do
  describe 'model relations' do
    it { is_expected.to belong_to(:example) }

    it { is_expected.to have_one(:example).dependent(:destroy) }
  end
end
```

File: spec/models/user_spec.rb [hunk 7]
```diff
@@ -1,32 +1,15 @@
...
-      end
+  describe 'associations' do
+    it { is_expected.to have_many(:tasks) }
+
+    it 'destroys dependent tasks after deletion' do
+      expect(Task.count).to equal(0)
+      user = create_user_with_tasks(tasks_count: 10)
+      expect(Task.count).to equal(10)
+      user.destroy
+      expect(Task.count).to equal(0)
```
Comment (StanisLoveSid) [inline spec/models/user_spec.rb:12]: same as above

File: spec/requests/api/v1/tasks/delete_spec.rb [hunk 8]
```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+RSpec.describe 'DELETE /v1/tasks/:id', type: :request do
```
Comment (StanisLoveSid) [inline spec/requests/api/v1/tasks/delete_spec.rb:3]: fix all failing request tests please

**Review by StanisLoveSid — CHANGES_REQUESTED** [review]

File: spec/factories/tasks.rb (hunk 5 above)
Comment (kirill-oleynik) [reply spec/factories/tasks.rb]: Oops! Seems like [FactoryBot gem'sdocumentation](https://github.com/thoughtbot/factory_bot/blob/master/GETTING_STARTED.md#has_many-associations) has another opinion. Why do U think that traits is better for this purpose? Lets discuss.

File: app/uploaders/application_uploader.rb [hunk 9]
```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+# Includes logic & behaviour required by all other uploaders
+class ImageUploader < Shrine
+  Shrine.plugin :activerecord
+  Shrine.plugin :validation_helpers
+  Shrine.plugin :determine_mime_type
```
Comment (StanisLoveSid) [inline app/uploaders/application_uploader.rb:7]:
```suggestion
  plugin :activerecord
  plugin :validation_helpers
  plugin :determine_mime_type
```

File: app/uploaders/image_uploader.rb [hunk 10]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+# Describes uploading image validation rules
+class ImageUploader < Shrine
+  Shrine.plugin :cached_attachment_data
+  Shrine.plugin :restore_cached_data
```
Comment (StanisLoveSid) [inline app/uploaders/image_uploader.rb:6]: same as above

File: app/uploaders/application_uploader.rb [hunk 11]
```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+# Includes logic & behaviour required by all other uploaders
+class ImageUploader < Shrine
```
Comment (StanisLoveSid) [inline app/uploaders/application_uploader.rb:4]: naming mistype

File: config/initializers/shrine.rb [hunk 12]
```diff
@@ -0,0 +1,5 @@
+require 'shrine'
+
+Shrine.plugin :activerecord
+Shrine.plugin :validation_helpers
+Shrine.plugin :determine_mime_type
```
Comment (StanisLoveSid) [inline config/initializers/shrine.rb:5]: can be moved to upploader

**Review by StanisLoveSid — APPROVED** [review]

**Pull request merged**